cs-google/src/main/scala/io/cloudslang/content/google/services/compute/disks/DiskController.scala

```diff
@@ -29,43 +29,37 @@ object DiskController {
     computeDisk
   }
 
-  def createAttachedDisk(boot:Boolean, autoDelete: Boolean, deviceNameOpt: Option[String], mode: String, source: String, interface: String): AttachedDisk = {
+  def createAttachedDisk(boot: Boolean,
+                         autoDelete: Boolean,
+                         mountMode: String,
+                         sourceOpt: Option[String] = None,
+                         deviceNameOpt: Option[String] = None,
+                         interfaceOpt: Option[String] = None,
+                         initializeParamsOpt: Option[AttachedDiskInitializeParams] = None): AttachedDisk = {
     val attachedDisk = new AttachedDisk()
       .setBoot(boot)
       .setAutoDelete(autoDelete)
-      .setMode(mode)
-      .setSource(source)
-      .setInterface(interface)
+      .setMode(mountMode)
 
-    deviceNameOpt match {
-      case Some(deviceName) => attachedDisk.setDeviceName(deviceName)
-      case _ => attachedDisk
+    sourceOpt match {
```

Comment:
if you don't want to group all the optionals (keep one/ match) together, it's more elegant to use:  `sourceOpt.foreach(source => attachedDisk.setSource(source))`  ... in general when you do pattern matching you should be exhaustive